### PR TITLE
[DO NOT MERGE] Bug 1170235 compat tables

### PR DESCRIPTION
**Pull request by stephaniehobson**
How to test:
- download this branch and compile your style sheets and compress assets
- create a macro from this: https://gist.github.com/darkwing/6fccdfd9852f338799eb
- create a page with the macro on it

Still to do:
- [ ] Icon font not appearing (may be unrelated issue)
- [x] Space in the table slides open but the history drawer itself just appears.
- [ ] Space in the table slides closed but the history drawer itself just dissipears.
- [x] The heading is also 'sliding open' when the history draw is opened in tablet/flex view. (I'll look at a fix)
- [x] Need explicit way to trigger thick border between support cells when medium switches.
- [x] Add `.bc-medium-border` to the macro

Where should we identify problems with the macro? The contents, legend, and title of many of the icons/abbreviations need to be changed.

### CODE REVIEW

**Comment by groovecoder** [comment]
Need to change the commit message on https://github.com/mozilla/kuma/commit/bea68fc1d442a910af27d747170e3d21997e74d9 to start with "Bug 1170235"

**Comment by darkwing** [comment]
Weird, the .woff is 403'ing.

**Comment by darkwing** [comment]
When expanding a cell to see its history, its "v" arrow seems to slide up.  This is due to the following elsewhere in kuma's CSS:

```
.text-content table {
border-collapse: collapse
}
```

File: media/stylus/components/compat-tables/bc-table.styl [hunk 1]
```diff
@@ -0,0 +1,308 @@
+/*
+table layout
+- uses display: block for smaller screen sizes;
+- requires .bc-table-{$num} modifier to display columns correctly
+====================================================================== */
+
+table.bc-table { /* specificity to over-ride .text-content styles */
```
Comment (darkwing) [inline media/stylus/components/compat-tables/bc-table.styl]: This isn't working, unfortunately, needs to be `.text-content table.bc-table {`

**Comment by darkwing** [comment]
Add a commit which I believe fixes item #2 in Stephanie's list.

**Comment by kleberveridiano** [comment]
ok!

2015-06-29 14:04 GMT-03:00 David Walsh <notifications@github.com>:

**Comment by stephaniehobson** [comment]
@darkwing It fixes the slide open, the same problem exists with the draw closing.

**Comment by stephaniehobson** [comment]
The font is still not displaying @groovecoder were you able to follow up on https://github.com/mozilla/kuma/pull/3248#issuecomment-111279718 ? Or should someone else take a look?

**Comment by stephaniehobson** [comment]
@darkwing Also, the page is jiggling around a lot during the expand and collapse but it doesn't on the prototype, could you look into that please?

**Comment by stephaniehobson** [comment]
I created the class `.bc-medium-border` to add to cells to trigger the thicker border between mediums since we can't expect that the same browser will always be in the first column the mobile side of the table. The macro will need to be updated to make use of it. @darkwing it's your macro do you want to do this work?

**Comment by openjck** [comment]
I notice we have a **[DO NOT MERGE]** tag on this. Can anyone say more about that? Does it still apply?

**Comment by stephaniehobson** [comment]
@openjck The other check boxes need check marks, I don't think we've identified specific people to fix them.

**Comment by darkwing** [comment]
Updated gist to accomplish the `bc-medium-border`:  https://gist.github.com/darkwing/46a4a72dea75f253113a

File: media/stylus/components/compat-tables/bc-table.styl [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+bc-table-tablet-display-flex() {
+    /*
+    Flexbox for those lucky few who can handle it
+    - we only needs this to make equal height columns in the tablet display view
+    - ie is not among those that can handle this flexbox code,
+      but it doesn't supoort @support so we need to include that in the MQ
+    - all browsers which support @support also use unprefixed syntax
+    ----------------------------------------------------------------------  */
+
+    .bc-table,
```
Comment (mdn-services) [inline media/stylus/components/compat-tables/bc-table.styl]: Flexible Box Layout Module not supported by: IE (10), IE Mobile (10)

File: media/stylus/components/compat-tables/bc-table.styl [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
```
Comment (mdn-services) [inline media/stylus/components/compat-tables/bc-table.styl]: Flexible Box Layout Module not supported by: IE (10), IE Mobile (10)

File: media/stylus/components/compat-tables/bc-table.styl [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+    .bc-table thead,
```
Comment (mdn-services) [inline media/stylus/components/compat-tables/bc-table.styl]: Flexible Box Layout Module not supported by: IE (10), IE Mobile (10)

File: media/stylus/components/compat-tables/bc-legend.styl [hunk 5]
```diff
@@ -0,0 +1,26 @@
+/*
+legend
+====================================================================== */
+
+.bc-legend {
+    vendorize(column-count, 2);
+    page-break-inside: avoid;
+    webkit-column-break-inside: avoid;
+    break-inside: avoid;
```
Comment (mdn-services) [inline media/stylus/components/compat-tables/bc-legend.styl]: CSS3 Multiple column layout not supported by: Firefox (37,38), Chrome (42,43), Safari (8,7.1), Opera (29,30), iOS Safari (8,8.1-8.3)

File: media/stylus/components/compat-tables/bc-legend.styl [hunk 6]
```diff
@@ -0,0 +1,28 @@
+/*
+legend
+====================================================================== */
+
+.bc-legend {
+    -moz-column-count: 2;
+    -webkit-column-count: 2;
+    column-count: 2;
```
Comment (mdn-services) [inline media/stylus/components/compat-tables/bc-legend.styl]: CSS3 Multiple column layout not supported by: Firefox (37,38), Chrome (42,43), Safari (8,7.1), Opera (29,30), iOS Safari (8,8.1-8.3)

File: media/stylus/components/compat-tables/bc-legend.styl (hunk 5 above)
Comment (openjck) [inline media/stylus/components/compat-tables/bc-legend.styl]: Hey, look at that!

Comment (groovecoder) [inline media/stylus/components/compat-tables/bc-legend.styl]: :dog: :pizza:

**Pull request closed without merging**